File: content-solutions/recursion/snowflake-knit.js
function solution(k) {
  function pattern(level) {
    if (level === 0) return ["*"];
    const small = pattern(level - 1);
    const blank = " ".repeat(small.length);
    const top = small.map((row) => row.repeat(3));
    const mid = small.map((row) => row + blank + row);
    return [...top, ...mid, ...top];
  }
  return pattern(k);
}

File: content-solutions/recursion/snowflake-knit.py
def solution(k):
    def pattern(level):
        if level == 0:
            return ["*"]
        small = pattern(level - 1)
        blank = " " * len(small)
        top = [row * 3 for row in small]
        mid = [row + blank + row for row in small]
        return top + mid + top

    return pattern(k)
